tools/buildgen/plugins/check_attrs.py

```diff
@@ -89,6 +89,13 @@ def subset_of(values):
         'vs_proj_dir': anything(),
         'zlib': one_of((True,)),
     },
+    'external_library': {
```

Comment:
you're calling this "external_library", but the only instances we currently plan to have are all "external proto libraries", so let's give some thought to how this should be named. Also the "proto_prefix" attributes signifies that this has something to do with protos.

From some other perspective, these are "external archives" to download rather than libraries.

Btw, we currently import the entire third_party/upb subtree into our git repo (because ubp isn't installable on its own so we embed it in our build), so perhaps we could consider making upb and "external library" as well.
CC @veblush

Reply:
Renamed to `external_proto_library` (feel free to suggest other names).

Yep, I do plan to look into the possibility of reducing our subtrees after the copied xds protos are gone. I recall one of the blocker: supporting `Make` without subtree is hard.